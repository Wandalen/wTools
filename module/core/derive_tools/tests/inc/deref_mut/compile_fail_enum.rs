//! # Test Matrix for `DerefMut` on Enums (Compile-Fail)
//!
//! This matrix documents test cases for ensuring the `DerefMut` derive macro correctly
//! rejects enums, as `DerefMut` is only applicable to structs with a single field.
//!
//! | ID   | Item Type | Expected Error Message                                   |
//! |------|-----------|----------------------------------------------------------|
//! | CF1.1 | Enum      | "DerefMut cannot be derived for enums. It is only applicable to structs with a single field." |

extern crate derive_tools_meta;

#[ allow( dead_code ) ]
#[ derive( derive_tools_meta::DerefMut ) ]
enum MyEnum
{
  Variant1( bool ),
  Variant2( i32 ),
}

fn main() {}
