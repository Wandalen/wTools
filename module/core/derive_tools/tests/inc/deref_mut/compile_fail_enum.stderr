error: DerefMut cannot be derived for enums. It is only applicable to structs with a single field.
  --> tests/inc/deref_mut/compile_fail_enum.rs:12:1
   |
12 | / #[ allow( dead_code ) ]
13 | | #[ derive( derive_tools_meta::DerefMut ) ]
14 | | enum MyEnum
...  |
17 | |   Variant2( i32 ),
18 | | }
   | |_^
